aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,621 @@
+
+Expression names
+^^^^^^^^^^^^^^^^^^^^^
+
+The ``asp-for`` attribute value is a `ModelExpression <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Rendering/ModelExpression/index.html>`__ and the right hand side of a lambda expression. Therefore, ``asp-for="Property1"`` becomes ``m => m.Property1`` in the generated code which is why you don't need to prefix with ``Model``.
+
+Collections
+^^^^^^^^^^^^^^
+
+Sample, a model containing an array of ``Colors``:
+
+.. literalinclude::  forms/sample/final/ViewModels/Person.cs
+  :language: c#
+  :lines: 5-10
+  :dedent: 3
+  :emphasize-lines: 4
+
+The action method:
+
+.. code-block:: c#
+
+    public IActionResult Edit(int id, int colorIndex)
+    {
+        ViewData["Index"] = colorIndex;        
+        return View(GetPerson(id));
+    }
+
+The following Razor shows how you access a specific ``Color`` element:
+
+.. literalinclude::   forms/sample/final/Views/Demo/EditColor.cshtml 
+  :language: HTML
+
+The *Views/Shared/EditorTemplates/String.cshtml* template:
+
+.. literalinclude::   forms/sample/final/Views/Shared/EditorTemplates/String.cshtml 
+  :language: HTML
+  
+Sample using ``List<T>``:
+
+.. literalinclude::  forms/sample/final/ViewModels/ToDoItem.cs
+  :language: c#
+  :lines: 3-7
+  :dedent: 3
+
+The following Razor shows how to iterate over a collection:
+
+.. literalinclude::   forms/sample/final/Views/Demo/Edit.cshtml 
+  :language: HTML
+
+Using the *Views/Shared/EditorTemplates/ToDoItem.cshtml* template reduces evaluating the indexer from three to one.
+
+.. literalinclude::   forms/sample/final/Views/Shared/EditorTemplates/ToDoItem.cshtml 
+  :language: HTML
+  
+:Note: Always use ``for`` (and *not* ``foreach``) to iterate over a list. Evaluating an indexer in a LINQ expression can be expensive and should be minimized.
+:Note: The commented sample code above shows how you would replace the lambda expression with the ``@`` operator to access each ``ToDoItem`` in the list.
+
+HTML Helper alternative to Input Tag Helper
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``Html.TextBox()``, ``Html.TextBoxFor()``, ``Html.Editor()`` and ``Html.EditorFor()`` have overlapping features with the Input Tag Helper. The Input Tag Helper will automatically set the ``type`` attribute; ``Html.TextBoxFor()`` will not. ``Html.EditorFor()`` handles collections, complex objects and templates; the Input Tag Helper does not. The Input Tag Helper is strongly typed (it uses lambda expressions); ``Html.TextBox()`` and ``Html.Editor()`` are not (they use expression names).
+
+Navigating child properties 
```

Comment:
Mention property paths in **Expression names** and move this before **Collections**. It's simpler and should be introduced before collections.